internal/store/installation.go

```diff
@@ -163,23 +163,72 @@ func (sqlStore *SQLStore) applyInstallationFilter(builder sq.SelectBuilder, filt
 	return builder
 }
 
-// GetInstallationsCount returns the number of installations filtered by the deletedat
-// field
-func (sqlStore *SQLStore) GetInstallationsCount(includeDeleted bool) (int, error) {
-	builder := sq.Select("COUNT(*) as InstallationsCount").From("Installation")
+// GetInstallationsCount returns the number of installations filtered by the
+// deleteAt field.
+func (sqlStore *SQLStore) GetInstallationsCount(includeDeleted bool) (int64, error) {
+	var totalResult countResult
+	installationBuilder := sq.
+		Select("Count (*)").
+		From("Installation")
 	if !includeDeleted {
-		builder = builder.Where("DeleteAt = 0")
+		installationBuilder = installationBuilder.Where("DeleteAt = 0")
+	}
+	err := sqlStore.selectBuilder(sqlStore.db, &totalResult, installationBuilder)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to query for total installations")
+	}
+	totalCount, err := totalResult.value()
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to get count result of total installations")
+	}
+
+	return totalCount, nil
+}
+
+// GetInstallationsStatus returns status of all installations which aren't
+// deleted.
+func (sqlStore *SQLStore) GetInstallationsStatus() (*model.InstallationsStatus, error) {
```

Comment:
What do you think of adding the option to filter by cluster id? That would be useful to check how many installations we have in a cluster and the state?

Reply:
This could be done, but would require quite a bit of extra processing. We would need to lookup the cluster installations(s) belonging to each installation in order to correlate the installation to a cluster. Definitely worth considering for a follow-up I think!

Reply:
Yeah, I think it's a great addition to check the load of a specific cluster. We can also have a `cloud cluster load` that would return the number of installations that the cluster has, etc. I concur is for a follow-up ticket